### PR TITLE
QGIS Vector geometry - tables for alg parameters (part two)

### PR DESCRIPTION
**Pull request by havatv**
Goal: Use tables for the algorithm parameters.
The algorithms from the start to (and including "Promote to multipart") .

Ticket(s): #4429

- [x] Backport to LTR documentation is required

### CODE REVIEW

**Comment by DelazJ** [comment]
@havatv, I don't know how many are left but can I suggest you do not add more algs to this PR, please? Otherwise it will be a nightmare to seriously review these thousands of (kind of repetitive) lines. I assume you do the changes from top to bottom then just create a new PR for the new changes you do. As long as you don't edit the same lines within your branches, the merge will make it smoothly. Thanks.

**Comment by havatv** [comment]
OK @DelazJ. I will continue in a new PR.
This one is ready for review.
It should not brake anything, so merging now should be safe.  It is possible to review afterwards (I am always ready to fix errors).

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 1]
```diff
@@ -1773,11 +2458,50 @@ Parameters
 ``Input layer`` [vector: any]
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 2]
```diff
@@ -1773,11 +2458,50 @@ Parameters
...
   Vector layer in input to extract the vertices from.
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 3]
```diff
@@ -1773,11 +2458,50 @@ Parameters
...
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
```

**Comment by havatv** [comment]
To avoid locking this part of the documentation for too long, I merged.
Reviews are welcome! I will still be able to fix error and make improvements.

**Comment by DelazJ** [comment]
Not a good idea! At all, imho! Three days i'm incrementally reviewing this PR, too big to be done in a row and, not fun if I may. A merged PR is harder to review because the suggestion mechanism is no longer available, hence back to the old days you have to fully write your suggestions. It was not fun before, i'm afraid it becomes boring... and i'll certainly go the lazy route.
Besides, I sincerely wonder who/what would be locked here. For weeks, you are the only one (at 98%) writing in this chapter; how could you lock yourself?

**Comment by havatv** [comment]
Sorry, I did not think about the inconvenience of not having the suggestion mechanism.
When it comes to locking, I am sad to admit that you are right - very few people are contributing.  But in addition to my table for parameters work, @jonnyforestGIS has been doing some work on check validity and error messages in this file. And we don't know what other contributors are considering to work with.  And since the changes that are made in these PRs are mostly cosmetic, not much harm could be done - so the consequences of late review should not be severe.
Sorry again for causing you trouble. I hope that I have not made too many mistakes in the cut and paste process here.

**Comment by jonnyforestGIS** [comment]
Hi @havatv and @DelazJ for now I'm not able to add more descriptions. I did a request in twitter to help to describe the missing geometries errors but at this time I have a lots of likes, retweets and some answers with examples that already I add. 
Understanding more this process of pulling, reviewing and merging now I  realise that can be easier to make all the changes or small changings and make de PR to do a single review instead of doing what I did in geometry validity.

I have a question @DelazJ we have some place such as mailing list, chatroom (e.g. irc, telegram, gitnode) where people that want to write documentation and say something like " I guys, I want to document this tool, is somebody doing this?"

Thank for your effort, I hope in the future to contribute more because I think documentation is essential for this project.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 4]
```diff
@@ -996,67 +1047,152 @@ polygons in the output layer for each group.
...
+     - Description
+   * - **Input layer**
+     - ``INPUT``
+     - [vector: any]
+     - Input vector layer
+   * - **New geometry type**
+     - ``TYPE``
+     - [enumeration]
+
+       Default: The first in the list
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Changed to "0", as normal for enumerations.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 5]
```diff
@@ -1416,197 +1832,449 @@ that happens to be processed.
...
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Z/M Dropped**
+     - ``OUTPUT``
+     - [same as input]
+     - The output vector layer (identical to the input layer,
+       except that M and/or Z values have been removed from
+       the geometries).
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: The dimensions are dropped, not just the values. I will update the text.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 6]
```diff
@@ -1705,57 +2375,100 @@ Outputs
...
+       vertices from the input layer geometries.
 
 
 .. _qgisextractvertices:
 
 Extract vertices
 ----------------
-Takes a line or polygon layer and generates a point layer with points representing
-the vertices in the input lines or polygons.
+Takes a line or polygon layer and generates a point layer with points
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: True. For point layers, the points are returned.

**Comment by havatv** [comment]
I have created a new PR (#4685) for the updates after your review.  I can keep that one open until you are satisfied.

**Comment by DelazJ** [comment]
>I have a question @DelazJ we have some place such as mailing list, chatroom (e.g. irc, telegram, gitnode) where people that want to write documentation and say something like " I guys, I want to document this tool, is somebody doing this?"

@jonnyforestGIS I think the mailing-list can be a good place to claim your interest for writing on a particular feature. If I'm not wrong, this is how you proceeded. About other channels, sorry I'm not on them and am not aware of anything. But at some point, since the writer will have to create a PR in this repository, informing us in here so that we  "assign" related issues to the person would be a nice step also imho (and avoid duplicate work).
> Understanding more this process of pulling, reviewing and merging now I realise that can be easier to make all the changes or small changings and make de PR to do a single review instead of doing what I did in geometry validity.

Then claim it to the world :smiley: , share your knowledge so that people are aware that git is just a "new" thing, not a hard thing. And contributing to docs is(/can be?) easy. You, new or occasional contributors to docs are the best advocates for the docs writing process. And if you think some things would have been better done on our side or were not clear, do not hesitate to share, either.

>Thank for your effort, I hope in the future to contribute more because I think documentation is essential for this project.

You are more than welcome. And I agree with you that we need a full featured and up to date docs for QGIS.

**Pull request merged**